### ISSUE TITLE
[num_seqs] computation problem for no packing case

### ISSUE THREAD

**Issue by Hevans123** [alibaba/Pai-Megatron-Patch#533]
https://github.com/alibaba/Pai-Megatron-Patch/blob/0bdda8277e6faf21402a956472e482600b9eacbe/megatron_patch/data/utils.py#L390-L392
There is some confusion. For "per_seq_average=True" and "reset_position_ids=False"， should "num_seqs" be 1 actually?
ie.
`subseq=loss_mask.sum(dim=-1).long()`
`loss_mask = loss_mask / subseq.view(-1,1)`
`num_seqs = torch.ones(position_ids.shape[0], device=torch.cuda.current_device(), dtype=torch.int64)`
If I misunderstood, please correct me.

**Comment by lostkevin** [comment]
Thanks for reporting. I'll create a PR to solve the issue
